libs/langchain/langchain/vectorstores/redis/base.py

```diff
@@ -864,7 +868,42 @@ def similarity_search(
         Returns:
             List[Document]: A list of documents that are most similar to the query
                 text.
+        """
+        query_embedding = self._embeddings.embed_query(query)
+        return self.similarity_search_by_vector(
+            query_embedding,
+            k=k,
+            filter=filter,
+            return_metadata=return_metadata,
+            distance_threshold=distance_threshold,
+            **kwargs,
+        )
+            
+    def similarity_search_by_vector(
+        self,
+        query_vector: List[float],
```

Comment:
I think ``vector`` makes more sense unless it's defined in the parent class

Reply:
Actually called `embedding` in the parent. I'll fix that.